chirp/drivers/tdh8.py

```diff
@@ -906,7 +893,47 @@ def get_memory(self, number):
         elif int(scan_val_list[number - 1]) == 1:
             mem.skip = ''
 
-        mem.mode = _mem.wide and "NFM" or "FM"
+        if self._is_txinh(_mem):
+                mem.duplex = ""
+                mem.offset = 0
+        elif int(_mem.rxfreq) == int(_mem.txfreq):
+            mem.duplex = ""
+            mem.offset = 0
+        elif abs(int(_mem.rxfreq) * 10 - int(_mem.txfreq) * 10) > 70000000:
+            mem.duplex = "split"
+            mem.offset = int(_mem.txfreq) * 10
+        else:
+            mem.duplex = int(_mem.rxfreq) > int(_mem.txfreq) and "-" or "+"
+            mem.offset = abs(int(_mem.rxfreq) - int(_mem.txfreq)) * 10
+
+        # Offset
+        if self._gmrs:
+            mem.duplex = ""
+            mem.offset = 0
+            mem.immutable = ['duplex']
+            if mem.number >= 1 and mem.number <= 30:
+                mem.immutable = ['freq']
```

Comment:
This will overwrite L913. You probably want `mem.immutable.append('freq')` right?